kong/status/ready.lua

```diff
@@ -0,0 +1,62 @@
+local declarative = require "kong.db.declarative"
+local constants = require "kong.constants"
+
+local tonumber = tonumber
+local kong = kong
+
+local get_current_hash = declarative.get_current_hash
+
+
+local worker_count = ngx.worker.count()
+local kong_shm = ngx.shared.kong
+local is_traditional = kong.configuration.database ~= "off"
+
+local DECLARATIVE_PLUGINS_REBUILD_COUNT_KEY = 
+                                constants.DECLARATIVE_PLUGINS_REBUILD_COUNT_KEY
+local DECLARATIVE_ROUTERS_REBUILD_COUNT_KEY =
+                                constants.DECLARATIVE_ROUTERS_REBUILD_COUNT_KEY
+local DECLARATIVE_EMPTY_CONFIG_HASH = constants.DECLARATIVE_EMPTY_CONFIG_HASH
+
+local function is_ready()
+
+  local ok = kong.db:connect()
+  if not ok then
+    return false
+  end
+
+  if is_traditional then
```

Comment:
You must check that the router is built even in this mode.
If not, why not?
please add a code comment explaining why this is done.

Reply:
> You must check that the router is built even in this mode. If not, why not? please add a code comment explaining why this is done.

In traditional mode, we don't know the router config in this mode because it is not fetched (from database) until user access the route. So it's impossible to forecast the router is built or not.